process/block/gasConsumption_test.go

```diff
@@ -630,13 +630,204 @@ func TestGasConsumption_ZeroIncomingLimit(t *testing.T) {
 	require.Zero(t, pendingMBs)      // added all
 }
 
+func TestGasConsumption_RevertIncomingMiniBlocks(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty mini block hashes should early exit", func(t *testing.T) {
+		t.Parallel()
+
+		gc, _ := block.NewGasConsumption(getMockArgsGasConsumption())
+		require.NotNil(t, gc)
+
+		gc.RevertIncomingMiniBlocks(nil)
+	})
+
+	t.Run("revert non-pending mini block should decrease total gas consumed", func(t *testing.T) {
+		t.Parallel()
+
+		gc, _ := block.NewGasConsumption(getMockArgsGasConsumption())
+		require.NotNil(t, gc)
+
+		// add mini blocks that will be within limits
+		mbs := generateMiniBlocks(3, 5)
+		txs := generateTxsForMiniBlocks(mbs)
+		lastMbIndex, pendingMbs, err := gc.CheckIncomingMiniBlocks(mbs, txs)
+		require.NoError(t, err)
+		require.Equal(t, 2, lastMbIndex)
+		require.Zero(t, pendingMbs)
+
+		totalGasBeforeRevert := gc.TotalGasConsumed()
+		expectedGasPerMb := maxGasLimitPerTx * 5 // 5 txs per mini block
+
+		// revert the first mini block and one none existing for coverage
+		gc.RevertIncomingMiniBlocks([][]byte{mbs[0].GetHash(), []byte("non-existent-hash")})
+
+		// total gas should decrease by the gas consumed by the first mini block
+		require.Equal(t, totalGasBeforeRevert-expectedGasPerMb, gc.TotalGasConsumed())
+
+		// revert the second and third mini blocks
+		gc.RevertIncomingMiniBlocks([][]byte{mbs[1].GetHash(), mbs[2].GetHash()})
+
+		// total gas should be zero now
+		require.Equal(t, uint64(0), gc.TotalGasConsumed())
+	})
+
+	t.Run("revert pending mini block at first position", func(t *testing.T) {
+		t.Parallel()
+
+		gc, _ := block.NewGasConsumption(getMockArgsGasConsumption())
+		require.NotNil(t, gc)
+
+		// add more mini blocks than the limit to create pending ones
+		mbs := generateMiniBlocks(10, 5)
+		txs := generateTxsForMiniBlocks(mbs)
+		lastMbIndex, pendingMbs, err := gc.CheckIncomingMiniBlocks(mbs, txs)
+		require.NoError(t, err)
+		require.Equal(t, 7, lastMbIndex)
+		require.Equal(t, 2, pendingMbs)
+
+		// get pending mini blocks
+		pendingMiniBlocks := gc.GetPendingMiniBlocks()
+		require.Len(t, pendingMiniBlocks, 2)
+
+		// revert the first pending mini block
+		firstPendingHash := mbs[8].GetHash()
+		gc.RevertIncomingMiniBlocks([][]byte{firstPendingHash})
+
+		// check that pending mini blocks were updated
+		updatedPendingMiniBlocks := gc.GetPendingMiniBlocks()
```

Comment:
can you validate also the updated totalGas in the revert tests as not all of them are currently doing it

Reply:
pushed